RecoParticleFlow/PFProducer/plugins/PFEGammaProducer.cc

```diff
@@ -104,35 +102,13 @@ PFEGammaProducer::produce(edm::Event& iEvent,
   iSetup.get<ESChannelStatusRcd>().get(esChannelStatusHandle_);
   pfeg_->setESChannelStatus(esChannelStatusHandle_.product());
 
-  // Get The vertices from the event
-  // and assign dynamic vertex parameters
-  edm::Handle<reco::VertexCollection> vertices;
-  bool gotVertices = iEvent.getByToken(vertices_,vertices);
-  if(!gotVertices) {
-    std::ostringstream err;
-    err<<"Cannot find vertices for this event.Continuing Without them ";
-    edm::LogError("PFEGammaProducer")<<err.str()<<std::endl;
-  }
-
   //Assign the PFAlgo Parameters
-  setPFVertexParameters(vertices.product());
+  setPFVertexParameters(iEvent.get(vertices_));
 
   // get the collection of blocks 
 
-  edm::Handle< reco::PFBlockCollection > blocks;
-
   LOGDRESSED("PFEGammaProducer")<<"getting blocks"<<std::endl;
-  bool found = iEvent.getByToken( inputTagBlocks_, blocks );  
```

Comment:
This would be a good opportunity to use the `auto handle = iEvent.getHandle(token_).makeValidHandle();` as @makortel suggested. Then, the following checks would truly unnecessary. Do you already have a specific implementation is mind for `makeValidHandle()`?